common/test/keyboards/build.sh

```diff
@@ -29,7 +31,8 @@ builder_parse "$@"
 function zipsource() {
   local target="$1"
   pushd "$1" > /dev/null
-  7z a -r -x!build -x"!$target.kpj.user" "${target}_source.zip" .
+  ZIP_FLAGS=("-q" "-r") # quiet, recursive
+  add_zip_files "${target}_source.zip" "-x@../zip-excludes" "${ZIP_FLAGS[@]}" .
```

Comment:
If introducing a new variable in the function, make it local.

```suggestion
  local ZIP_FLAGS=("-q" "-r") # quiet, recursive
  add_zip_files "${target}_source.zip" "-x@../zip-excludes" "${ZIP_FLAGS[@]}" .
```

But, why split ZIP_FLAGS out? This seems more complicated than it needs to be.

Also, for clarity, normally, flag parameters would not be quoted unless they contained special characters that made that necessary.

```suggestion
  add_zip_files "${target}_source.zip" -x@../zip-excludes -q -r . # -q quiet, -r recursive
```